xicam/plugins/__init__.py

```diff
@@ -266,9 +272,20 @@ def loadPlugins(self):
         # Remove candidates list since we don't need them any more and
         # don't need to take up the space
         delattr(self, "_candidates")
+
+        # Load entry points
+        self.load_entry_point_plugins()
+
         return self.processed_plugins
 
-    def load_plugin(self, candidate_infofile, candidate_filepath, plugin_info):
+    def load_entry_point_plugins(self):
+        for category_name, plugins in self.category_mapping.items():
+            entry_point_plugins = [EntryPointPluginInfo(entry_point) for entry_point in
+                                   pkg_resources.iter_entry_points(f'xicam.plugins.{category_name}')]
```

Comment:
The `pkg_resources` implementation is problematic. I am not familiar with the details, but I have seen this come up a couple times. My impression is that it's just very slow, and may not cope with eggs well (?). I know that Min RK (of IPython) stated that they wouldn't have considered using entrypoints in the IPython/Jupyter ecosystem if it weren't for the superior implementation in [`entrypoints`](https://entrypoints.readthedocs.io/). You can use `entrypoints.get_group_named(...)` in place of `pkg_resources.iter_entry_points(...)`. It returns a dict of `Entrypoint` objects. Those should drop right into you existing code; they have a `load()` method as well.

Note that if you want to handle potential name collisions you can use `entrypoints.get_group_all(...)` which returns a _list_ instead of a dict. It might be convenient to [steal this pattern from databroker](https://github.com/bluesky/databroker/blob/effedde941b0b8153bd99216f86be79971b03f46/databroker/core.py#L1106-L1117) for collision-checking.

Reply:
Huh, it's disappointing that the vanilla solution is lacking. Looks like it should be easy to swap out though. Thanks!